doc/api/addons.markdown

```diff
@@ -1,39 +1,47 @@
 # Addons
 
-Addons are dynamically-linked shared objects. They can provide glue to C and
-C++ libraries. The API (at the moment) is rather complex, involving
-knowledge of several libraries:
+Node.js Addons are dynamically-linked shared objects, written in C or C++, that
+can be loaded into Node.js using the `require()` statement, and used just as if
+they were an ordinary Node.js module. They are used primarily to provide an
+interface between JavaScript running in Node.js and C/C++ libraries.
 
- - V8 JavaScript, a C++ library. Used for interfacing with JavaScript:
-   creating objects, calling functions, etc.  Documented mostly in the
+At the moment, The method for implementing Addons is rather complicated,
+involving knowledge of several components and APIs :
+
+ - V8 JavaScript: the C++ library Node.js currently uses to provide the
+   JavaScript implementation. V8 provides the mechanisms for creating objects,
+   calling functions, etc. V8's API is documented mostly in the
    `v8.h` header file (`deps/v8/include/v8.h` in the Node.js source
    tree), which is also available [online][].
 
- - [libuv][], C event loop library. Anytime one needs to wait for a file
-   descriptor to become readable, wait for a timer, or wait for a signal
-   to be received, one will need to interface with libuv. That is, if you
-   perform any I/O, libuv will need to be used.
-
- - Internal Node.js libraries. The most important class is `node::ObjectWrap`
-   which you will likely want to derive from.
+ - [libuv][]: The C library that implements the Node.js event loop. Anytime
```

Comment:
Also need to amend this section to include _asynchronous operations_ because that's another job of libuv for addons—if you want async, you have to go through this beasty. libuv can also be used to provide cross-platform implementations of add-ons in many cases.

How about:

> The C library that implements the Node.js event loop, its worker threads and all of the asynchronous behaviors of the platform. It also serves as a cross-platform abstraction library, giving easy, POSIX-like access across all major operating systems to many common system tasks, such as interacting with the filesystem, sockets, timers and system events. libuv also provides a pthreads-like threading abstraction that may be used to power more sophisticated asynchronous addons that need to move beyond the standard event loop. Addon authors are encouraged to think about how to avoid blocking the event loop with I/O or other time-intensive tasks by off-loading work via libuv to non-blocking system operations, worker threads or a custom use of libuv's threads.